Index: linux-amlogic-3.10-bringup/include/linux/amlogic/osd/osd_dev.h
===================================================================
--- linux-amlogic-3.10-bringup.orig/include/linux/amlogic/osd/osd_dev.h	2014-12-02 10:43:26.000000000 +0800
+++ linux-amlogic-3.10-bringup/include/linux/amlogic/osd/osd_dev.h	2014-12-02 14:18:39.496616049 +0800
@@ -28,13 +28,14 @@
 #include "osd.h"
 #include <linux/amlogic/vout/vinfo.h>
 #include <linux/amlogic/logo/logo.h>
-
+#include "../../../../../hardware/arm/gpu/ump/include/ump/ump_kernel_interface.h"
 #define  OSD_COUNT 	2 /* we have two osd layer on hardware*/
 
 #define KEYCOLOR_FLAG_TARGET  1
 #define KEYCOLOR_FLAG_ONHOLD  2
 #define KEYCOLOR_FLAG_CURRENT 4
 
+#define OSD_MAX_FB 2
 typedef struct myfb_dev {
     struct mutex lock;
 
@@ -54,6 +55,9 @@
 	u32  preblend_enable;
 	u32  enable_key_flag;
 	u32  color_key;	
+#ifdef CONFIG_FB_AMLOGIC_UMP
+       ump_dd_handle ump_wrapped_buffer[OSD_MAX_FB][2];
+#endif
 } myfb_dev_t;
 typedef  struct list_head   list_head_t ;
 
@@ -129,7 +133,7 @@
 #if defined (CONFIG_FB_OSD2_CURSOR)
 extern void osddev_cursor(struct myfb_dev *fbdev, s16 x, s16 y, s16 xstart, s16 ystart, u32 osd_w, u32 osd_h);
 #endif
-
+extern  int osddev_copy_data_tocursor(myfb_dev_t *g_fbi, aml_hwc_addr_t *cursor_mem);
 extern  void  osddev_set_colorkey(u32 index,u32 bpp,u32 colorkey );
 extern  void  osddev_srckey_enable(u32  index,u8 enable);
 extern void  osddev_set_gbl_alpha(u32 index,u32 gbl_alpha) ;
