Index: arm-src-kernel-2015-01-15-321cfb5a46/Makefile
===================================================================
--- arm-src-kernel-2015-01-15-321cfb5a46.orig/Makefile	2015-01-13 10:04:47.000000000 +0800
+++ arm-src-kernel-2015-01-15-321cfb5a46/Makefile	2015-01-19 17:23:28.294979779 +0800
@@ -373,17 +373,13 @@
 		   -fno-strict-aliasing -fno-common \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
-		   -fno-delete-null-pointer-checks \
-		   -Werror
+		   -fno-delete-null-pointer-checks
 KBUILD_CFLAGS   += -Werror=enum-compare \
 		   -Werror=comment \
 		   -Werror=implicit-int \
 		   -Werror=missing-braces \
 		   -Werror=unused-value \
-		   -Werror=maybe-uninitialized \
-		   -Werror=unused-variable \
 		   -Werror=format \
-		   -Werror=unused-function \
 		   -Werror=switch \
 		   -Werror=strict-prototypes \
 		   -Werror=declaration-after-statement \
@@ -394,6 +390,7 @@
 		   -Werror=return-type \
 		   -Werror=parentheses \
 		   -Werror=int-to-pointer-cast \
+		   -Wno-maybe-uninitialized \
 		   -Wno-error=cpp
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
