Index: linux-amlogic-3.10-bringup/include/linux/amlogic/osd/osd_hw.h
===================================================================
--- linux-amlogic-3.10-bringup.orig/include/linux/amlogic/osd/osd_hw.h	2014-12-02 10:43:26.000000000 +0800
+++ linux-amlogic-3.10-bringup/include/linux/amlogic/osd/osd_hw.h	2014-12-02 14:18:39.536616049 +0800
@@ -240,6 +240,7 @@
 #if defined(CONFIG_FB_OSD2_CURSOR)
 extern void osd_cursor_hw(s16 x, s16 y, s16 xstart, s16 ystart, u32 osd_w, u32 osd_h, int index);
 #endif
+extern void osddev_copy_data_tocursor_hw(u32 cursor_mem_paddr, aml_hwc_addr_t *hwc_mem);
 extern void osd_suspend_hw(void);
 extern void osd_resume_hw(void);
 extern void osd_init_hw(u32  logo_loaded);
