Index: aml_nand-2015-01-15-6ad64c2a7a/amlnf/Makefile
===================================================================
--- aml_nand-2015-01-15-6ad64c2a7a.orig/amlnf/Makefile	2014-12-25 20:36:30.000000000 +0800
+++ aml_nand-2015-01-15-6ad64c2a7a/amlnf/Makefile	2015-01-15 15:54:56.328396453 +0800
@@ -37,7 +37,7 @@
 aml_nftl_dev-objs += block/aml_nftl_hw_interface.o
 #aml_nftl_dev-objs += ../../../../../../../../../hardware/amlogic/nand/amlnf/nftl/aml_nftl_core_20141024.o
 #aml_nftl_dev-objs += ../../../../../../../../../hardware/amlogic/nand/amlnf/nftl/aml_nftl_core.o
-aml_nftl_dev-objs += ../../../../../../../../../hardware/amlogic/nand/amlnf/nftl/aml_nftl_core_20141222.o_shipped
+aml_nftl_dev-objs += nftl/aml_nftl_core_20141222.o_shipped
 
 #aml_nftl_dev-objs += nftl/aml_nftl_phy.o       
 #aml_nftl_dev-objs  += nftl/aml_nftl_list.o      
