Index: elementary-1.15.3/src/lib/elm_widget_prefs.h
===================================================================
--- elementary-1.15.3.orig/src/lib/elm_widget_prefs.h	2015-08-26 01:46:02.000000000 +0800
+++ elementary-1.15.3/src/lib/elm_widget_prefs.h	2016-02-16 16:22:58.170634056 +0800
@@ -70,7 +70,7 @@
 /**
  * Base widget smart data extended with prefs instance data.
  */
-typedef struct _Elm_Prefs_Data Elm_Prefs_Data;
+// typedef struct _Elm_Prefs_Data Elm_Prefs_Data;
 struct _Elm_Prefs_Data
 {
    Elm_Prefs_Page_Node  *root;
