diff --git a/drivers/amlogic/mmc/emmc_partitions.c b/drivers/amlogic/mmc/emmc_partitions.c
index 1925423..e35b19e 100644
--- a/drivers/amlogic/mmc/emmc_partitions.c
+++ b/drivers/amlogic/mmc/emmc_partitions.c
@@ -503,7 +503,7 @@ static int add_emmc_partition(struct gendisk *disk,
 	return 0;
 }
 
-static int is_card_emmc(struct mmc_card *card)
+int is_card_emmc(struct mmc_card *card)
 {
 	struct mmc_host *mmc = card->host;
 
diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index fd4928c..aa6107b 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -2071,6 +2071,8 @@ static inline int mmc_blk_readonly(struct mmc_card *card)
 	       !(card->csd.cmdclass & CCC_BLOCK_WRITE);
 }
 
+int is_card_emmc(struct mmc_card *card);
+
 static struct mmc_blk_data *mmc_blk_alloc_req(struct mmc_card *card,
 					      struct device *parent,
 					      sector_t size,
@@ -2099,7 +2101,10 @@ static struct mmc_blk_data *mmc_blk_alloc_req(struct mmc_card *card,
 	 * index anymore so we keep track of a name index.
 	 */
 	if (!subname) {
-		md->name_idx = find_first_zero_bit(name_use, max_devices);
+		if (!is_card_emmc(card))
+			md->name_idx  = find_first_zero_bit(name_use, max_devices);
+		else
+			md->name_idx  = 1;
 		__set_bit(md->name_idx, name_use);
 	} else
 		md->name_idx = ((struct mmc_blk_data *)
