--- gst-plugins-good-0.10.31/gst/isomp4/qtdemux_fourcc.h.orig	2011-12-30 05:59:09.000000000 -0800
+++ gst-plugins-good-0.10.31/gst/isomp4/qtdemux_fourcc.h	2014-08-12 17:26:16.686123371 -0700
@@ -140,6 +140,10 @@ G_BEGIN_DECLS
 #define FOURCC_drmi     GST_MAKE_FOURCC('d','r','m','i')
 #define FOURCC_avc1     GST_MAKE_FOURCC('a','v','c','1')
 #define FOURCC_avcC     GST_MAKE_FOURCC('a','v','c','C')
+#define FOURCC_H265     GST_MAKE_FOURCC('H','2','6','5')
+#define FOURCC_hev1     GST_MAKE_FOURCC('h','e','v','1')
+#define FOURCC_hvc1     GST_MAKE_FOURCC('h','v','c','1')
+#define FOURCC_hvcC     GST_MAKE_FOURCC('h','v','c','C')
 #define FOURCC_btrt     GST_MAKE_FOURCC('b','t','r','t')
 #define FOURCC_VP31     GST_MAKE_FOURCC('V','P','3','1')
 #define FOURCC_rle_     GST_MAKE_FOURCC('r','l','e',' ')
