--- brcmap6xxx-ap6xxx/broadcm_40181/Kconfig.orig	2014-04-16 14:21:34.435769702 +0800
+++ brcmap6xxx-ap6xxx/broadcm_40181/Kconfig	2014-04-16 14:21:46.511769930 +0800
@@ -10,21 +10,21 @@ config SDIO_DHD_CDC_WIFI_40181_MODULE
 config BCMDHD_FW_PATH
 	depends on SDIO_DHD_CDC_WIFI_40181_MODULE
 	string "Firmware path"
-	default "/system/etc/wifi/fw_bcmdhd.bin"
+	default "/etc/wifi/fw_bcmdhd.bin"
 	---help---
 	  Path to the firmware file.
 
 config BCMDHD_NVRAM_PATH
 	depends on SDIO_DHD_CDC_WIFI_40181_MODULE
 	string "NVRAM path"
-	default "/system/etc/wifi/nvram.txt"
+	default "/etc/wifi/nvram.txt"
 	---help---
 	  Path to the calibration file.
 
 config BCMDHD_CONFIG_PATH
 	depends on SDIO_DHD_CDC_WIFI_40181_MODULE
 	string "Config path"
-	default "/system/etc/wifi/config.txt"
+	default "/etc/wifi/config.txt"
 	---help---
 	  Path to the driver configuration file.
 
