--- webkitgtk-1.11.5/Source/WebCore/GNUmakefile.list.am.orig	2014-06-03 11:43:43.846688228 +0800
+++ webkitgtk-1.11.5/Source/WebCore/GNUmakefile.list.am	2014-06-03 11:43:58.082688494 +0800
@@ -6171,11 +6171,11 @@ webcoregtk_sources += \
 	Source/WebCore/plugins/gtk/gtk2xtbin.h \
 	Source/WebCore/plugins/gtk/PluginPackageGtk.cpp \
 	Source/WebCore/plugins/gtk/PluginViewGtk.cpp \
-	Source/WebCore/plugins/gtk/xembed.h
+	Source/WebCore/plugins/gtk/xembed.h \
+	Source/WebCore/platform/graphics/cairo/GLContext.cpp \
+	Source/WebCore/platform/graphics/cairo/GLContext.h 
 if USE_OPENGL
 webcoregtk_sources += \
-	Source/WebCore/platform/graphics/cairo/GLContext.cpp \
-	Source/WebCore/platform/graphics/cairo/GLContext.h \
 	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.cpp \
 	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.h
 endif # END USE_OPENGL
