Index: uboot-next/board/amlogic/configs/m6_mbx_v1.h
===================================================================
--- uboot-next.orig/board/amlogic/configs/m6_mbx_v1.h	2014-11-20 09:50:50.000000000 +0800
+++ uboot-next/board/amlogic/configs/m6_mbx_v1.h	2014-11-24 13:58:01.463625481 +0800
@@ -175,7 +175,7 @@
 	"display_color_bg=0\0" \
 	"fb_addr=0x85100000\0" \
 	"sleep_threshold=20\0" \
-	"upgrade_step=0\0" \
+	"upgrade_step=2\0" \
 	"batlow_threshold=10\0" \
 	"batfull_threshold=98\0" \
 	"outputmode=720p\0" \
@@ -191,7 +191,9 @@
 	"irremote_update=if irkey 0x41beb14e 500000 ;then run update; fi\0" \
 	"nand_key_burning=saradc open 4;if saradc get_in_range 0x164 0x1b4 ;then msleep 500;if saradc get_in_range 0x164 0x1b4; then tiny_usbtool 20000; fi; fi\0" \
 	"cvbscheck=setenv outputtemp ${outputmode};if test ${outputmode} = 480i; then if test ${cvbsenable} = true; then setenv outputtemp 480cvbs;fi;fi; if test ${outputmode} = 576i; then if test ${cvbsenable} = true; then setenv outputtemp 576cvbs;fi;fi\0" \
-	"nandboot=echo Booting from nand ...;run nandargs;nand read boot ${loadaddr} 0 600000;hdcp prefetch nand;bootm;run recovery\0" \
+	"nandboot=echo Booting from nand ...;run nandargs;nand read boot ${loadaddr} 0 600000;hdcp prefetch nand;bootm\0" \
+    "bootsdcard=echo Booting ...;run nandargs;run bootsdargs;mmcinfo;ext2load mmc 0 ${loadaddr} uImage;hdcp prefetch nand;bootm\0" \
+    "bootupdate=echo Booting ...;run nandargs;run bootupdateargs;mmcinfo;fatload mmc 0 ${loadaddr} uImage;hdcp prefetch nand;bootm\0" \
 	"nand_recovery=echo enter recovery;run nandargs;if mmcinfo; then if fatload mmc 0 ${loadaddr} recovery.img; then bootm;fi;fi; if usb start; then if fatload usb 0 ${loadaddr} recovery.img; then bootm;fi;fi; nand read recovery ${loadaddr} 0 600000; bootm\0" \
 	"mmcboot=echo Booting from mmc ...;run mmcargs;mmcinfo 1; mmc read 1 ${loadaddr} ${mmc_boot_offset} ${mmc_lk_size};hdcp prefetch emmc; bootm; run recovery\0" \
 	"mmc_recovery=echo enter recovery;run mmcargs;if mmcinfo; then if fatload mmc 0 ${loadaddr} recovery.img; then bootm;fi;fi; if usb start; then if fatload usb 0 ${loadaddr} recovery.img; then bootm;fi;fi; mmcinfo 1; mmc read 1 ${loadaddr} ${mmc_recovery_offset} ${mmc_lk_size}; bootm\0" \
@@ -218,18 +220,21 @@
 		"run nand_key_burning; run updatekey_or_not; run irremote_update; run switch_bootmode\0" \
 	"nandargs="\
 	        "run cvbscheck;"\
-	        "setenv bootargs root=/dev/cardblksd2 rw rootfstype=ext3 rootwait init=/init console=ttyS0,115200n8 logo=osd1,loaded,${fb_addr},${outputtemp},full androidboot.resolution=${outputmode} hdmimode=${hdmimode} cvbsmode=${cvbsmode} hlt vmalloc=256m mem=1024m a9_clk_max=1512000000 vdachwswitch=${vdacswitchmode} hdmitx=${cecconfig}\0"\
+            "setenv bootargs root=/dev/system rw rootfstype=ext4 rootwait init=/sbin/init console=ttyS0,115200n8 logo=osd1,loaded,${fb_addr},${outputtemp},full androidboot.resolution=${outputmode} hdmimode=${hdmimode} cvbsmode=${cvbsmode} hlt vmalloc=256m mem=1024m a9_clk_max=1512000000 vdachwswitch=${vdacswitchmode} hdmitx=${hdmimode}\0"\
 	"mmcargs="\
 		"run cvbscheck;"\
 		"mmcinfo 1;"\
-		"setenv bootargs root=/dev/cardblksd2 rw rootfstype=ext3 rootwait init=/init console=ttyS0,115200n8 logo=osd1,loaded,${fb_addr},${outputtemp},full androidboot.resolution=${outputmode} nohlt vmalloc=256m mem=1024m a9_clk_max=1512000000 vdachwswitch=${vdacswitchmode} hdmitx=${cecconfig}\0"\
+		"setenv bootargs root=/dev/cardblksd1 rw rootfstype=ext2 rootwait init=/sbin/init console=ttyS0,115200n8 logo=osd1,loaded,${fb_addr},${outputtemp},full androidboot.resolution=${outputmode} nohlt vmalloc=256m mem=1024m a9_clk_max=1512000000 vdachwswitch=${vdacswitchmode} hdmitx=${cecconfig}\0"\
 	"bootargs="\
-	    "root=/dev/cardblksd2 rw rootfstype=ext3 rootwait init=/init console=ttyS0,115200n8 nohlt no_console_suspend vmalloc=256m mem=1024m logo=osd1,loaded,${fb_addr},720p\0" \
+	    "root=/dev/cardblksd1 rw rootfstype=ext2 rootwait init=/sbin/init console=ttyS0,115200n8 nohlt no_console_suspend vmalloc=256m mem=1024m logo=osd1,loaded,${fb_addr},720p\0" \
     "switch_bootmode="\
 		"if test ${reboot_mode} = factory_reset;"\
 			"then run recovery;"\
 		"else if test ${reboot_mode} = update;"\
 			"then run recovery;fi;fi\0" \
+    "bootsdargs=setenv bootargs root=/dev/cardblksd1 rw rootfstype=ext2 rootwait init=/sbin/init console=ttyS0,115200n8 logo=osd1,0x85100000,${outputtemp},full hdmimode=${hdmimode} cvbsmode=${cvbsmode} hlt vmalloc=256m mem=1024m a9_clk_max=1512000000 vdachwswitch=${vdacswitchmode} hdmitx=${hdmimode}\0"\
+    "bootupdateargs=setenv bootargs root=/dev/cardblksd1 rw rootfstype=vfat rootwait init=/sbin/init console=ttyS0,115200n8 logo=osd1,0x85100000,${outputtemp},full hdmimode=${hdmimode} cvbsmode=${cvbsmode} hlt vmalloc=256m mem=1024m a9_clk_max=1512000000 vdachwswitch=${vdacswitchmode} hdmitx=${hdmimode} firmware=rootfs.tar.gz\0"\
+
 
 #define CONFIG_BOOTCOMMAND \
  "setenv bootcmd run compatible_boot; saveenv; run compatible_boot; run compatible_boot"
@@ -275,7 +280,7 @@
 	"switch_bootmode=if test ${reboot_mode} = factory_reset; then run recovery;else if test ${reboot_mode} = update; then run recovery;fi;fi\0" \
 	"nandboot=echo Booting from nand ...;run nandargs;nand read boot ${loadaddr} 0 600000;hdcp prefetch nand;bootm;run recovery\0" \
 	"recovery=echo enter recovery;run nandargs;if mmcinfo; then if fatload mmc 0 ${loadaddr} recovery.img; then bootm;fi;fi;if usb start; then if fatload usb 0 ${loadaddr} recovery.img; then bootm;fi;fi; nand read recovery ${loadaddr} 0 600000; bootm\0" \
-	"bootargs=root=/dev/cardblksd2 rw rootfstype=ext3 rootwait init=/init console=ttyS0,115200n8 nohlt vmalloc=256m mem=1024m logo=osd1,0x84100000,720p\0" \
+	"bootargs=root=/dev/cardblksd2 rw rootfstype=ext3 rootwait init=/sbin/init console=ttyS0,115200n8 nohlt vmalloc=256m mem=1024m logo=osd1,0x84100000,720p\0" \
 	"usbnet_devaddr=00:15:18:01:81:31" \
 	"usbnet_hostddr=00:15:18:01:a1:3b" \
 	"cdc_connect_timeout=9999999999" \
