Index: uboot-next/board/amlogic/configs/m3_f16_no_video_v1.h
===================================================================
--- uboot-next.orig/board/amlogic/configs/m3_f16_no_video_v1.h	2014-11-20 09:50:50.000000000 +0800
+++ uboot-next/board/amlogic/configs/m3_f16_no_video_v1.h	2014-11-24 13:58:01.407625480 +0800
@@ -109,7 +109,7 @@
 	"boardname=m1_mbox\0" \
 	"chipname=8726m\0" \
 	"machid=B8E\0" \
-	"bootargs=root=/dev/cardblksd2 rw rootfstype=ext2 rootwait init=/init console=ttyS0,115200n8 nohlt a9_clk=600M clk81=200M mem=512m\0" \
+	"bootargs=root=/dev/mmcblk0p1 rw rootfstype=ext2 rootwait init=/sbin/init console=ttyS0,115200n8 nohlt a9_clk=600M clk81=200M mem=512m\0" \
 	"partnum=2\0" \
 	"p0start=1000000\0" \
 	"p0size=400000\0" \
@@ -123,7 +123,7 @@
 	"normalstart=1000000\0" \
 	"normalsize=400000\0" \
 
-#define CONFIG_BOOTCOMMAND  "mmcinfo;fatload mmc 0:1 82000000 uimage-m3;bootm"
+#define CONFIG_BOOTCOMMAND  "mmcinfo;ext2load mmc 0:1 82000000 boot.img;bootm"
 
 #define CONFIG_AUTO_COMPLETE	1
 
