Index: MPlayer-1.2/ffmpeg/libavcodec/aarch64/Makefile
===================================================================
--- MPlayer-1.2.orig/ffmpeg/libavcodec/aarch64/Makefile	2015-10-04 05:12:06.000000000 +0800
+++ MPlayer-1.2/ffmpeg/libavcodec/aarch64/Makefile	2015-12-08 15:50:35.194452603 +0800
@@ -13,7 +13,7 @@
 OBJS-$(CONFIG_VC1_DECODER)              += aarch64/vc1dsp_init_aarch64.o
 OBJS-$(CONFIG_VORBIS_DECODER)           += aarch64/vorbisdsp_init.o
 
-ARMV8-OBJS-$(CONFIG_VIDEODSP)           += aarch64/videodsp.o
+OBJS-$(CONFIG_VIDEODSP)           += aarch64/videodsp.o
 
 NEON-OBJS-$(CONFIG_FFT)                 += aarch64/fft_neon.o
 NEON-OBJS-$(CONFIG_H264CHROMA)          += aarch64/h264cmc_neon.o
