--- gstreamer-0.10.36/plugins/elements/gstinputselector.c.orig	2012-01-05 10:15:35.000000000 -0800
+++ gstreamer-0.10.36/plugins/elements/gstinputselector.c	2014-04-27 20:02:24.593448800 -0700
@@ -1492,17 +1492,33 @@ gst_input_selector_activate_sinkpad (Gst
 {
   GstPad *active_sinkpad;
   GstSelectorPad *selpad;
+	gchar *padname = NULL;
+	gint ret = 0;
 
   selpad = GST_SELECTOR_PAD_CAST (pad);
 
   selpad->active = TRUE;
   active_sinkpad = sel->active_sinkpad;
   if (active_sinkpad == NULL) {
+		#if 0
     /* first pad we get activity on becomes the activated pad by default */
     if (sel->active_sinkpad)
       gst_object_unref (sel->active_sinkpad);
     active_sinkpad = sel->active_sinkpad = gst_object_ref (pad);
     GST_DEBUG_OBJECT (sel, "Activating pad %s:%s", GST_DEBUG_PAD_NAME (pad));
+		#else
+		/*sink0 we get activity on becomes the activated pad*/
+		if (sel->active_sinkpad)
+      gst_object_unref (sel->active_sinkpad);
+		
+		padname = GST_STR_NULL(GST_OBJECT_NAME (pad));
+		if(NULL != padname) {
+			ret = g_strcmp0(padname, "sink0");
+			if(0 == ret) {
+				active_sinkpad = sel->active_sinkpad = gst_object_ref (pad);
+			}
+		}
+		#endif
   }
 
   return active_sinkpad;
