Index: gpu-r4p0-01/ump/Kbuild
===================================================================
--- gpu-r4p0-01.orig/ump/Kbuild	2014-09-12 14:32:18.000000000 +0800
+++ gpu-r4p0-01/ump/Kbuild	2014-09-23 13:58:03.911128661 +0800
@@ -42,29 +42,19 @@
 ccflags-y += -DMALI_STATE_TRACKING=0
 ccflags-y += -DMALI_ENABLE_CPU_CYCLES=0
 ccflags-$(CONFIG_UMP_DEBUG) += -DDEBUG
+ccflags-$(CONFIG_MALI400_DEBUG) += -DDEBUG
 
 # For customer releases the Linux Device Drivers will be provided as ARM proprietary and GPL releases:
 # The ARM proprietary product will only include the license/proprietary directory
 # The GPL product will only include the license/gpl directory
 
 ifeq ($(wildcard $(TOP_KBUILD_SRC)$(src)/linux/license/gpl/*),)
-ccflags-y += -I$(TOP_KBUILD_SRC)$(src)/linux/license/proprietary
+ccflags-y += -I$(TOP_KBUILD_SRC)$(src)/linux/license/proprietary -I$(TOP_KBUILD_SRC)$(src)/$(UDD_FILE_PREFIX)linux/license/proprietary
 else
-ccflags-y += -I$(TOP_KBUILD_SRC)$(src)/linux/license/gpl
+ccflags-y += -I$(TOP_KBUILD_SRC)$(src)/linux/license/gpl -I$(TOP_KBUILD_SRC)$(src)/$(UDD_FILE_PREFIX)linux/license/gpl
 endif
 
 
-$(obj)/__mali_osk_atomics.c : $(src)/$(UDD_FILE_PREFIX)linux/mali_osk_atomics.c
-	@cp -f $< $@
-$(obj)/__mali_osk_locks.c : $(src)/$(UDD_FILE_PREFIX)linux/mali_osk_locks.c
-	@cp -f $< $@
-$(obj)/__mali_osk_memory.c : $(src)/$(UDD_FILE_PREFIX)linux/mali_osk_memory.c
-	@cp -f $< $@
-$(obj)/__mali_osk_math.c : $(src)/$(UDD_FILE_PREFIX)linux/mali_osk_math.c
-	@cp -f $< $@
-$(obj)/__mali_osk_misc.c : $(src)/$(UDD_FILE_PREFIX)linux/mali_osk_misc.c
-	@cp -f $< $@
-
 ump-y = common/ump_kernel_common.o \
 	common/ump_kernel_descriptor_mapping.o \
 	common/ump_kernel_api.o \
@@ -78,23 +68,12 @@
 	linux/ump_osk_atomics.o \
 	linux/ump_osk_low_level_mem.o \
 	linux/ump_osk_misc.o \
-	__mali_osk_locks.o \
-	__mali_osk_memory.o \
-	__mali_osk_atomics.o \
-	__mali_osk_math.o \
-	__mali_osk_misc.o 
-
-
-
-
-#	$(UDD_FILE_PREFIX)linux/mali_osk_atomics.o \
-#	$(UDD_FILE_PREFIX)linux/mali_osk_locks.o \
-#	$(UDD_FILE_PREFIX)linux/mali_osk_memory.o \
-#	$(UDD_FILE_PREFIX)linux/mali_osk_math.o \
-#	$(UDD_FILE_PREFIX)linux/mali_osk_misc.o
-
+	$(UDD_FILE_PREFIX)linux/mali_osk_atomics.o \
+	$(UDD_FILE_PREFIX)linux/mali_osk_locks.o \
+	$(UDD_FILE_PREFIX)linux/mali_osk_memory.o \
+	$(UDD_FILE_PREFIX)linux/mali_osk_math.o \
+	$(UDD_FILE_PREFIX)linux/mali_osk_misc.o
 
-clean-files := __mali_osk*.c 
 
 obj-$(CONFIG_UMP) := ump.o
 
